runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -3396,6 +3198,728 @@ J9::Z::TreeEvaluator::ArrayCopyBNDCHKEvaluator(TR::Node * node, TR::CodeGenerato
+      indirectReadWriteBarrierHelperForFieldWatch(node, owningMethod, bcIndex, true /* isWrite */, cg);
+      }
+   }
+
+void J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   // The child of rdbar for static field is for side effects and is
+   // not used by the store evaluator
+   TR::Node *sideEffectNode = node->getSymbolReference()->getSymbol()->isStatic() ? node->getFirstChild() : NULL;
+   if (cg->comp()->getOption(TR_EnableFieldWatch))
+      {
+      TR::TreeEvaluator::VMrdbarEvaluatorForFieldWatch(node, cg);
+      if (sideEffectNode)
+         {
+         cg->decReferenceCount(sideEffectNode);
+         }
+      }
+   else if (sideEffectNode)
+      {
+      cg->recursivelyDecReferenceCount(sideEffectNode);
+      }
+   }
+
+TR::Register *
+J9::Z::TreeEvaluator::irdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::iloadEvaluator(node, cg);
+   } 
+
+TR::Register *
+J9::Z::TreeEvaluator::frdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::floadEvaluator(node, cg);
+   } 
+
+TR::Register *
+J9::Z::TreeEvaluator::drdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::dloadEvaluator(node, cg);
+   }
+
+TR::Register *
+J9::Z::TreeEvaluator::ardbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::aloadEvaluator(node, cg);
+   }
+
+TR::Register *
+J9::Z::TreeEvaluator::brdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::bloadEvaluator(node, cg);
+   }
+
+TR::Register *
+J9::Z::TreeEvaluator::srdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::sloadEvaluator(node, cg);
+   }
+
+TR::Register *
+J9::Z::TreeEvaluator::lrdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   J9::Z::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   return TR::TreeEvaluator::lloadEvaluator(node, cg);
+   }
+
+void
+J9::Z::TreeEvaluator::wrtbarSideEffectsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   // The 2nd/3rd child of wrtbar for static/instance field is for side affect
+   // and is not used by store evaluator
+   TR::Node *sideEffectNode = node->getSymbolReference()->getSymbol()->isStatic() ? node->getSecondChild(): node->getThirdChild();
+   if (cg->comp()->getOption(TR_EnableFieldWatch))
+      {
+      TR::TreeEvaluator::VMwrtbarEvaluatorForFieldWatch(node, cg);
+      cg->decReferenceCount(sideEffectNode);
+      }
+   else
+      {
+      cg->recursivelyDecReferenceCount(sideEffectNode);
```

Comment:
So by default wrtbar has no side effect. This is added as a stub in case someone uses wrtbar for another purpose but hasn't implemented the evaluator yet.